src/tests/multihost/alltests/test_misc.py

```diff
@@ -532,3 +541,81 @@ def test_0009_dbus_method_find_usrby_attr(multihost, backupsssdconf, ldap_posix_
+            9. Add a 50ms delay to the network interface using the 'tc' command.
+        :steps:
+            1. Measures the response time for the 'getent netgroup netgrp_nowait'
+                command and saves it to the 'res_time' variable.
+            2. Run a loop that repeats the 'getent netgroup netgrp_nowait' command 4 times
+                and checks if the response time is less than to the initial response time.
+            3. Wait for 15 seconds before deleting the contents of the '/var/log/sssd/sssd_nss.log' file again.
+            4. Remove the network delay added in step 9 using the 'tc' command.
+        :expectedresults:
+            1. res_time variable will have the response time
+            2. Response time is less than to the initial response time
+            3. Wait for 15 seconds
+            4. Network delay should be removed
+        """
+        client = multihost.client[0]
+        intf = [i for i in
+                client.run_command("nmcli device status | "
+                                   "grep connected | "
+                                   "awk '{print $1}'").stdout_text.split("\n") if i != ""]
+        ldap_uri = 'ldap://%s' % (multihost.master[0].sys_hostname)
+        ds_rootdn = 'cn=Directory Manager'
+        ds_rootpw = 'Secret123'
+        ldap_inst = LdapOperations(ldap_uri, ds_rootdn, ds_rootpw)
+        ldap_inst.org_unit("Netgroup", ds_suffix)
+        user_dn = f'cn=netgrp_nowait,ou=Netgroup,{ds_suffix}'
+        user_info = {'cn': 'netgrp_nowait'.encode('utf-8'),
+                     'objectClass': ['nisNetgroup'.encode('utf-8'),
+                                     'top'.encode('utf-8')],
+                     'nisNetgroupTriple': '(host1,kau10,example.com)'.encode('utf-8')}
+        ldap_inst.add_entry(user_info, user_dn)
+        tools = sssdTools(multihost.client[0])
+        tools.sssd_conf("nss", {'filter_groups': 'root',
+                                'filter_users': 'root',
+                                'debug_level': '9',
+                                'entry_cache_nowait_percentage': '50'}, action='update')
+        tools.sssd_conf("domain/example1",
+                        {'entry_cache_timeout': '30',
+                         'ldap_netgroup_search_base': f"ou=Netgroup,{ds_suffix}"}, action='update')
+        tools.clear_sssd_cache()
+        client.run_command("> /var/log/sssd/sssd_nss.log")
+        client.run_command(f"tc qdisc add dev {intf[0]} root netem delay 50ms")
+        start = float("%.20f" % time.time())
+        client.run_command("getent netgroup netgrp_nowait")
+        end = float("%.20f" % time.time())
+        res_time = end - start
+        time.sleep(16)
+        for i in range(4):
+            start = float("%.20f" % time.time())
+            client.run_command("getent netgroup netgrp_nowait")
+            end = float("%.20f" % time.time())
+            loop_response = end - start
+            assert loop_response < res_time
```

Comment:
Store the loop responses in a list instead and add assert after the test cleans up.
Do it similar way for the find_logs, just store the found/not found as bools 
```
response_times = []
log_message_present = []
for ....
     ...
    response_times.append(loop_response)
    log_message_present.append(find_logs(multihost, log_sssd, "Performing midpoint cache update of   [netgrp_nowait@example1]")
    ...
   
 assert all([ bool( i <= res_time)  for i in  response_times]), "what went wrong"
 assert all(log_message_present), "what went wrong"
``` 

Or just store the result as bools using logical operators, just make sure to put the asserts in the end and do all of the proper cleanup before.

```
passed = True
for ...
    ...
    passed = passed and ( loop_response < res_time ) and find_logs(multihost, log_sssd, "Performing midpoint cache update of [netgrp_nowait@example1]")
    ...
    
assert passed, "what went wrong"
```